soroban-spec/fixtures/ts/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import * as SorobanClient from 'soroban-client';
+import { xdr } from 'soroban-client';
+import { Buffer } from "buffer";
+import { scValStrToJs } from './convert';
+import { invoke, InvokeArgs } from './invoke';
+
+export * from './constants'
+export * from './server'
+export * from './invoke'
+
+export type u32 = number;
+export type i32 = number;
+export type u64 = bigint;
+export type i64 = bigint;
+export type u128 = bigint;
+export type i128 = bigint;
+export type u256 = bigint;
+export type i256 = bigint;
+export type Address = string;
+export type Option<T> = T | undefined;
+
+/// Error interface containing the error message
+export interface Error_ { message: string };
+
+export interface Result<T, E = Error_> {
+    unwrap(): T,
+    unrap_err(): E,
+    isOk(): boolean,
+    isErr(): boolean,
+};
+
+export class Ok<T> implements Result<T> {
+    constructor(readonly value: T) { }
+    unrap_err(): Error_ {
+        throw new Error('No error');
+    }
+    unwrap(): T {
+        return this.value;
+    }
+
+    isOk(): boolean {
+        return true;
+    }
+
+    isErr(): boolean {
+        return !this.isOk()
+    }
+}
+
+export class Err<T> implements Result<T> {
+    constructor(readonly error: Error_) { }
+    unrap_err(): Error_ {
+        return this.error;
+    }
+    unwrap(): never {
+        throw new Error(this.error.message);
+    }
+
+    isOk(): boolean {
+        return false;
+    }
+
+    isErr(): boolean {
+        return !this.isOk()
+    }
+}
+
+if (typeof window !== 'undefined') {
+    //@ts-ignore Buffer exists
+    window.Buffer = window.Buffer || Buffer;
+}
+
+const regex = /ContractError\((\d+)\)/;
+
+function get_error(err: string): Err<Error_> | undefined {
+    const match = err.match(regex);
+    if (!match) {
+        return undefined;
+    }
+    // @ts-ignore
+    let i = parseInt(match[1], 10);
+    if (i < Errors.length) {
+        return new Err(Errors[i]!);
+    }
+    return undefined;
+}
+
+export async function put({key, val}: {key: string, val: string}): Promise<void> {
+    let invokeArgs: InvokeArgs = {method: 'put', args: [((i) => xdr.ScVal.scvSymbol(i))(key),
+        ((i) => xdr.ScVal.scvSymbol(i))(val)], };
+    
+    // @ts-ignore Type does exist
+    const response = await invoke(invokeArgs);
+    return ;
+    
+    
```

Comment:
nitpick: weird whitespace, needs a formatter pass

Reply:
oh these are test fixtures :facepalm:

Reply:
Removed the extra spaces!